Frameworks/CoreText/CTFrame.mm

```diff
@@ -111,16 +111,23 @@ void CTFrameGetLineOrigins(CTFrameRef frameRef, CFRange range, CGPoint origins[]
 /**
  @Status Interoperable
 */
-void CTFrameDraw(CTFrameRef frame, CGContextRef ctx) {
-    uint32_t count = [((_CTFrame*)frame)->_lines count];
-    CGPoint curTextPos = CGContextGetTextPosition(ctx);
+void CTFrameDraw(CTFrameRef frameRef, CGContextRef ctx) {
+    _CTFrame* frame = static_cast<_CTFrame*>(frameRef);
     if (frame && ctx) {
-        CGPoint curTextPos = CGContextGetTextPosition(ctx);
-        for (_CTLine* line in static_cast<id<NSFastEnumeration>>(static_cast<_CTFrame*>(frame)->_lines)) {
-            CGPoint newPos = curTextPos + line->_lineOrigin;
-            CGContextSetTextPosition(ctx, newPos.x, newPos.y);
+        // Need to invert and translate coordinates so frame draws from top-left
+        CGContextSaveGState(ctx);
+        CGRect boundingRect;
+        _CGPathGetBoundingBoxInternal(frame->_path.get(), &boundingRect);
+        CGContextTranslateCTM(ctx, 0, boundingRect.size.height);
+        CGAffineTransform textMatrix = CGContextGetTextMatrix(ctx);
+        CGContextSetTextMatrix(ctx, CGAffineTransformScale(textMatrix, 1.0f, -1.0f));
+        CGContextScaleCTM(ctx, 1.0f, -1.0f);
+        for (_CTLine* line in static_cast<id<NSFastEnumeration>>(frame->_lines)) {
+            CGContextSetTextPosition(ctx, line->_lineOrigin.x, line->_lineOrigin.y);
             _CTLineDraw(static_cast<CTLineRef>(line), ctx, false);
         }
+
+        CGContextRestoreGState(ctx);
```

Comment:
or at least it shouldn't, according to the docs. idk about our current impl though lemme check

---
In reply to: [84372445](https://github.com/Microsoft/WinObjC/pull/1197#discussion_r84372445) [](ancestors = 84372445)